frontend/app/components/root/root.tsx

```diff
@@ -96,46 +94,64 @@ const messages = defineMessages({
   },
 });
 
-/** main component fr main comments widget */
-export class Root extends Component<Props, State> {
-  constructor(props: Props) {
-    super(props);
+function getInitialSort() {
+  const sort = localStorage.getItem(LS_SORT_KEY) as Sorting;
 
-    this.state = {
-      isLoaded: false,
-      isCommentsListLoading: false,
-      commentsShown: maxShownComments,
-      wasSomeoneUnblocked: false,
-    };
-
-    this.onBlockedUsersShow = this.onBlockedUsersShow.bind(this);
-    this.onBlockedUsersHide = this.onBlockedUsersHide.bind(this);
-    this.onUnblockSomeone = this.onUnblockSomeone.bind(this);
-    this.showMore = this.showMore.bind(this);
+  if (sort) {
+    return sort;
   }
 
-  async componentWillMount() {
-    Promise.all([this.props.fetchUser(), this.props.fetchComments(this.props.sort)]).finally(() => {
-      this.setState({
-        isLoaded: true,
-      });
+  return DEFAULT_SORT;
+}
+
+/** main component fr main comments widget */
+export class Root extends Component<Props, State> {
+  state = {
+    sort: getInitialSort(),
+    isUserLoading: true,
+    isCommentsListLoading: true,
+    commentsShown: maxShownComments,
+    wasSomeoneUnblocked: false,
+    isSettingsVisible: false,
+  };
+
+  componentWillMount() {
+    const userloading = this.props.fetchUser().finally(() => this.setState({ isUserLoading: false }));
 
+    Promise.all([userloading, this.fetchComments()]).finally(() => {
+      postMessage({ remarkIframeHeight: document.body.offsetHeight });
+      this.setState({ isCommentsListLoading: false });
       setTimeout(this.checkUrlHash);
       window.addEventListener('hashchange', this.checkUrlHash);
     });
 
     window.addEventListener('message', this.onMessage.bind(this));
   }
 
-  logIn = async (p: AuthProvider): Promise<User | null> => {
-    const user = await this.props.logIn(p);
-    await this.props.fetchComments(this.props.sort);
+  fetchComments(sort = this.state.sort) {
+    return this.props.fetchComments(sort);
```

Comment:
fetchComments() {
  const sort = this.state.sort;
  return this.props.fetchComments(sort);
}

Reply:
[This parameter used in this place](https://github.com/umputun/remark42/pull/609/files/c2f5bd44c3755491eb7894dd8c26f0da779745f1#diff-df6edac8c631ae459ffbd3dc621acd4bR138
)